app/models/host/base.rb

```diff
@@ -246,6 +246,10 @@ def overwrite=(value)
     end
 
     def primary_interface
+      # Returns a dummy object if host is unmanaged and without interfaces.
+      # Unmanaged hosts still have the interface requirement due to delegations and certain validations.
+      # A refactoring that allows unmanaged hosts to exist without interfaces would be welcome.
+      return Nic::Base.new(:host => self, :primary => true) unless managed && interfaces.present?
```

Comment:
We pass `:host => self` and  `:primary => true` so that [NameSynchronizer](https://github.com/theforeman/foreman/blob/develop/app/services/name_synchronizer.rb) sets the name of the host properly. Yes. I also think it couldn't be dirtier but *1.8* is near and we won't have another RC for testing.